### PR TITLE
Passthrough redirect service

### PR DESCRIPTION
**Pull request by allouis**
Got some code for us? Awesome 🎊!

Please include a description of your change & check your PR against this list, thanks!

- [ ] There's a clear use-case for this code change, explained below
- [ ] Commit message has a short title & references relevant issues
- [ ] The build will pass (run `yarn test:all` and `yarn lint`)

We appreciate your contribution! 

Also, if you'd be interested in writing code like this for us more regularly, we're hiring:
https://careers.ghost.org/product-engineer-node-js

### CODE REVIEW

File: ghost/link-replacement/lib/link-replacement.js [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
+
         // 1. Add attribution
         // TODO: this should move the the attribution service in the future
         url.searchParams.append('rel', newsletter.get('slug') + '-newsletter');
         url.searchParams.append('attribution_id', post.id);
         url.searchParams.append('attribution_type', 'post');
 
         // 2. Add redirect for link click tracking
-        const redirect = await this.createRedirect(url.toString());
+        const redirect = await this.#linkRedirectService.addRedirect(url);
```
Comment (SimonBackx) [inline ghost/link-replacement/lib/link-replacement.js:44]: If we add `from` as a parameter, what do you think if we would do something like this instead (doesn't have to be in this PR):
```
const redirect = await this.#linkClickTrackingService.createTrackedRedirect(url);
```
Which internally would call `await this.#linkRedirectService.addRedirect(generatedSlug, url);`

Comment (SimonBackx) [reply ghost/link-replacement/lib/link-replacement.js:44]: Maybe we can also pass the member uuid so that it includes that in the returned `from`

Comment (SimonBackx) [reply ghost/link-replacement/lib/link-replacement.js:44]: Essentially we could also name it `this.#linkClickTrackingService.createTrackedLink(url);` which returns a normal URL and hides the internal implementation detail that it works via redirects 🤷‍♂️

File: ghost/link-redirects/lib/LinkRedirectsService.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const LinkRedirect = require('./LinkRedirect');
+
+class LinkRedirectsService {
+    /**
+     * @param {URL} to
+     *
+     * @returns {Promise<LinkRedirect>}
+     */
+    async addRedirect(to) {
```
[the remark this answers is not in the document]
Comment (allouis) [reply ghost/link-redirects/lib/LinkRedirectsService.js:9]: As discussed will do this in a later commit

**Pull request merged**